### DISCUSSION TITLE
Esp8266 pwm 100% on startup /need Help

### DISCUSSION THREAD

**Discussion by Matten-Matten** [arendst/Tasmota#13496] [Q&A] [answered]
hello, I have described an ESP8266 with tasmota and connected a pwm circuit with an IRFZ44N to gpio12 and would like to use it to dim an LED strip. The dimming works so far. When the voltage is switched on, the LEDs light up briefly at 100% until the ESP is ready and dims the last saved value.

Question:
can you somehow prevent this from lighting up at the beginning?

DE:
hallo, ich habe ein ESP8266 mit tasmota beschrieben und eine pwm Schaltung mit einem IRFZ44N an gpio12 angeschlossen und möchte damit ein LED Stripe dimmen. Das dimmen funktioniert soweit. Beim zuschalten der Spannung leuchten die LEDs kurz auf 100% auf bis der ESP bereit ist und den zuletzt gespeicherten Wert dimmt.

Frage:
kann man das aufleuchten am Anfang irgendwie verhindern?

Gruß
Matten Matten

**Comment by s-hadinger** [comment, marked as answer]
Tasmota always starts PWM at low level (GND) but during boot the gpio is configured as input. Depending on electrical design, being not configured as output could cause the bright led. 
Is gpio12 configured as PWM or PWMi?

**Reply by joba-1** [reply]
you don't show the cirquit diagram, so this is a bit of guesswork ( and no guarantees anyways ;) )

If your N-mosfets source S is connected to ground and its gate G to gpio12/D6 then add a pull down resistor (~10k) between S and G or use gpio15/D8 instead of D6 which is pulled down internally already. 
This will make sure the light stays off at power up until tasmota can take over control.

**Reply by sfromis** [reply]
I'm reminded of the H801 led controller, which has a short bright flash after power up, outside the control of what Tasmota can do anything about. As the controller is not meant for switching on/off much, I tend to see this flash as a confirmation of it working.

**Reply by Matten-Matten** [reply]
It works with pull down resistor (~10k)!
Thank you for the fix answer.

matten matten

**Reply by Matten-Matten** [reply]
> Tasmota always starts PWM at low level (GND) but during boot the gpio is configured as input. Depending on electrical design, being not configured as output could cause the bright led. Is gpio12 configured as PWM or PWMi?

The gpio 12 configured as pwm 1
